### PR TITLE
Small test cleanups

### PR DESCRIPTION
**Pull request by martinpitt**
I noticed these while investigating the failures on rhel-9-0.

### CODE REVIEW

File: test/check-application [hunk 1]
```diff
@@ -885,13 +885,11 @@ class TestApplication(testlib.MachineCase):
...
     def testCheckpointRestoreCGroupsV2(self):
         b = self.browser
-        m = self.machine
 
         if not self.has_cgroupsV2:
             return
 
-        # Only Fedora works with CGroupsV2
-        if m.image.startswith('fedora'):
+        if self.has_criu:
```
Comment (martinpitt) [inline test/check-application:892]: @marusak: I'm not 100% sure about this change -- conceptually, hardcoding "fedora" here seems wrong and error prone. In practice, RHEL 9 has cgroups v2 and a recent criu (same as F34), but throws "does not support checkpoint/restore". But I'd consider that a bug (filing now), not a behavior to validate. It also does not help with `testCheckpointRestoreCGroupsV1` where rhel-9 fails in the same way, and this hack doesn't help us.

Comment (martinpitt) [reply test/check-application:892]: I filed https://bugzilla.redhat.com/show_bug.cgi?id=1944964 and I'm about to do a naughty for it -- let me know if you think that's the right course of action, or you'd rather hack the tests.

Comment (martinpitt) [reply test/check-application:892]: If they don't want to support criu in RHEL 9, then they should drop the `Recommends: criu`, and our image preparation can anticipate that with `rpm -e criu` and setting `has_criu` to False on RHEL-9.

**Comment by martinpitt** [comment]
Interesting - [fedora 34 packit](http://artifacts.dev.testing-farm.io/7f34aaee-3e7c-4666-acc0-09810cf6ff52/) failed `testCheckpointRestoreCGroupsV2`: The [screenshot](http://artifacts.dev.testing-farm.io/7f34aaee-3e7c-4666-acc0-09810cf6ff52//work-alldOdoeO/plans/all/discover/default/tests/TestApplication-testCheckpointRestoreCGroupsV2-fedora-33-127.0.0.1-22-FAIL.png) shows "requires at least CRIU 31100", which is usually the symptom of `criu` not being installed at all. I wonder how this PR broke that -- the previous `if m.image.startswith('fedora')` should have been true previously as well.

I added another commit to make sure criu is installed in FMF.

**Comment by martinpitt** [comment]
That wasn't it. Needs local reproducing/examination, moving to draft.

**Comment by martinpitt** [comment]
Re-dropping the "depend on criu" commit, it didn't help. Also unwinding the "Simplify testCheckpointRestoreCGroupsV2" to confirm if that really was the cause. The other commits really ought to be no-effective-change cleanups, but computers..

**Comment by martinpitt** [comment]
Meh, now I'm surprised -- it [failed again](http://artifacts.dev.testing-farm.io/bdb33e8e-35fa-463c-ba01-96174df91477/). So at least it was not the "Simplify testCheckpointRestoreCGroupsV2" commit. Post-holiday material now..

**Comment by martinpitt** [comment]
A-ha! The FMF packit test is generally broken on F33, it happened in #698 as well. So marking ready for review again, and this needs to be investigated separately.

**Review by marusak — APPROVED** [review]
Thanks!
Should this make rhel-9 green?

**Comment by martinpitt** [comment]
@marusak : It helps for rhel-9-0, but there is still some trouble with criu, see https://bugzilla.redhat.com/show_bug.cgi?id=1944964 . We said we wanted to wait until upstream acks the bug or says "sorry, not supported". But they assigned the bug now, so I guess let's treat that as a naughty? So we can land this, and I'll submit a bots PR with naughty + rhel-9 enablement, and I'm also looking into that cgroupsv2 test failure on TF fedora-33.

**Pull request merged**